src/main/kotlin/study/step2/Operation.kt

```diff
@@ -0,0 +1,27 @@
+package study.step2
+
+import java.util.Arrays
+import java.util.function.BiFunction
+
+enum class Operation(
+    val operation: String,
+    private val operationFunction: BiFunction<Double, Double, Double>
+) {
+    PLUS("+", BiFunction<Double, Double, Double> { a: Double, b: Double -> a + b }),
+    MINUS("-", BiFunction<Double, Double, Double> { a: Double, b: Double -> a - b }),
+    MULTIPLY("*", BiFunction<Double, Double, Double> { a: Double, b: Double -> a * b }),
+    DIVISION("/", BiFunction<Double, Double, Double> { a: Double, b: Double -> a / b });
+
+    fun calculateNumbers(num1: Double, num2: Double): Double {
+        return operationFunction.apply(num1, num2)
+    }
+
+    companion object {
+        fun chooseOperation(operation: String?): Operation {
+            return Arrays.stream(Operation.values())
+                .filter { v: Operation -> v.operation == operation }
+                .findAny()
+                .orElseThrow { IllegalArgumentException("일치하는 연산자가 없습니다.") }
+        }
```

Comment:
```suggestion
        fun chooseOperation(operation: String?): Operation {
            return Operation.values()
                .firstOrNull() { it.operation == operation }
                ?: throw IllegalArgumentException("일치하는 연산자가 없습니다.")
        }
```

맛보기이지만, kotlin collection 함수를 잘 사용한다면 이렇게도 표현이 가능합니다 :)

Reply:
오... 바로 탐색해서 일치하는 것을 반환할 수 있나보군요.
좋네요! 다음번에 활용해 볼게요!